templates/commands/render/action_include.go

```diff
@@ -126,3 +150,32 @@ func includePath(ctx context.Context, inc *spec.IncludePath, sp *stepParams) err
 	}
 	return nil
 }
+
+// TODO(#158): add support for ignoring file name. For example, "/foo" ignores
+// a single file named foo in the root, but "foo" ignores all files named foo
+// anywhere.
+func ignorePaths(ctx context.Context, sp *stepParams, fromDir string) (map[string]struct{}, error) {
```

Comment:
There's a different way this could work. I'm not sure it would be better, but we should consider it.

The way this code works, as written, is:
 1. We expand the ignore globs into a list of files using filepath.Glob(). This involves doing a bunch of filesystem operations.
 2. When we're deciding whether to ignore each file, we check if that filename is in the list of files returned from filepath.Glob()
 
This approach works, but arguably slower and more complicated than it needs to be. Step 1 doesn't actually need to happen and those slow-ish filesystem operations can be avoided.

Instead, we could use [filepath.Match()](https://pkg.go.dev/path/filepath#Match) instead of filepath.Glob(). Inside the Visitor function, we already have the filename of each file, and we're considering if it should be ignored. We can do that by iterating over each ignore glob and calling `filepath.Match(ignoreGlob, filename)`, ignoring the file if any glob returns true.

On the other hand, there is some nice simplicity in sharing code with processGlobs(), which argues in favor of the existing algorithm.

I'm leaning towards the filepath.Match() approach because I want this code to stay fast, and file IO is kinda slow.

Reply:
I see, filepath.Match() does sound better, let me try it. Thanks!